config/config_test.go

```diff
@@ -120,6 +120,40 @@ creation_rules:
       - 'https://baz.vault:8200/v1/baz/keys/baz-key'
 `)
 
+var sampleConfigWithMergeType = []byte(`
+creation_rules:
+  - path_regex: ""
+    key_groups:
+    - merge:
+      - kms:
+        - arn: foo
+          aws_profile: bar
+        pgp:
+        - bar
+        gcp_kms:
+        - resource_id: foo
+        azure_keyvault:
+        - vaultUrl: https://foo.vault.azure.net
+          key: foo-key
+          version: fooversion
+        hc_vault:
+        - 'https://foo.vault:8200/v1/foo/keys/foo-key'
+      - kms:
+        - arn: baz
+          aws_profile: foo
+        pgp:
+        - qux
+        gcp_kms:
+        - resource_id: bar
+        - resource_id: baz
+        azure_keyvault:
+        - vaultUrl: https://bar.vault.azure.net
+          key: bar-key
+          version: barversion
+        hc_vault:
+        - 'https://baz.vault:8200/v1/baz/keys/baz-key'
```

Comment:
Maybe add another keygroup to `merge` that itself has a `merge` key with more key groups.

Reply:
Will do.
I will rename the keys a little bit, so they are easier to see through.

Reply:
Please confirm that the updated test is more complete.